internal/util/GitService.go

```diff
@@ -395,14 +407,14 @@ func (impl GitServiceImpl) GetCloneDirectory(targetDir string) (clonedDir string
 func (impl GitServiceImpl) Clone(url, targetDir string) (clonedDir string, err error) {
 	impl.logger.Debugw("git checkout ", "url", url, "dir", targetDir)
 	clonedDir = filepath.Join(impl.config.GitWorkingDir, targetDir)
-	_, err = git.PlainClone(clonedDir, false, &git.CloneOptions{
-		URL:  url,
-		Auth: impl.Auth,
-	})
+	_, errorMsg, err := impl.gitCliUtil.Clone(clonedDir, url, impl.Auth.Username, impl.Auth.Password)
 	if err != nil {
-		impl.logger.Errorw("error in git checkout ", "url", url, "targetDir", targetDir, "err", err)
+		impl.logger.Errorw("error in git checkout", "url", url, "targetDir", targetDir, "err", err)
 		return "", err
 	}
+	if errorMsg != "" {
```

Comment:
move this code inside clone function

Reply:
the reason behind maintaining 2 diff error states 
err: represent error at azure end
errorMsg: error in api like repo doesn't exists, branch not found, no permission etc ...